File: app/src/main/java/spiderbiggen/shoppingcart/Data/Store.java
package spiderbiggen.shoppingcart.Data;

import io.realm.RealmObject;
import io.realm.annotations.Ignore;
import io.realm.annotations.PrimaryKey;
import io.realm.annotations.Required;

/**
 * Created by Stefan Breetveld on 14-3-2016.
 * Part of ShoppingCart.
 */
public class Store extends RealmObject {

    public static final String STORE_ID = "storeId";
    public static final String STORE_NAME = "storeName";


    @PrimaryKey
    private int storeId;

    private String storeName;

    public Store() {
    }

    public Store(int i, String string) {
        storeId = i;
        storeName = string;
    }

    public int getStoreId() {
        return storeId;
    }

    public void setStoreId(int storeId) {
        this.storeId = storeId;
    }

    public String getStoreName() {
        return storeName;
    }

    public void setStoreName(String storeName) {
        this.storeName = storeName;
    }
}

File: app/src/main/java/spiderbiggen/shoppingcart/data/Store.kt
package spiderbiggen.shoppingcart.data

import spiderbiggen.shoppingcart.data.interfaces.IItem
import spiderbiggen.shoppingcart.data.interfaces.IStore

/**
 * Created by Stefan Breetveld on 31-3-2016.
 * Part of ShoppingBasket.
 */
data class Store(override val id: Int, override var name: String, override val itemList: MutableList<IItem>) : IStore {

    companion object {
        var maxId: Int = 0
    }


    init {
        if(id > maxId) maxId = id
    }
    constructor(id: Int, storeName: String): this(id, storeName, mutableListOf())
    constructor(storeName: String) : this(++maxId, storeName)

    override fun compareTo(other: IStore): Int {
        return this.name.compareTo(other.name)
    }

    override fun toString() :String {
        return this.name
    }
}